templates/types/streaming/fastapi/app/tasks/llama_cloud.py

```diff
@@ -0,0 +1,89 @@
+import os
+import logging
+import requests
+from typing import List, Any, Dict, Optional
+
+
+logger = logging.getLogger("uvicorn")
+
+
+class LLamaCloudFile:
+
+    LLAMA_CLOUD_URL = "https://cloud.llamaindex.ai/api/v1"
+    # TODO: move to output/llamacloud later
+    LOCAL_STORE_PATH = "data/private"
+
+    DOWNLOAD_FILE_NAME_TPL = "{pipeline_id}${filename}"
+
+    @classmethod
+    def get_files(cls, pipeline_id: str) -> List[Dict[str, Any]]:
+        url = f"{cls.LLAMA_CLOUD_URL}/pipelines/{pipeline_id}/files"
+        return cls.make_request(url)
+
+    @classmethod
+    def get_file_detail(cls, project_id: str, file_id: str) -> Dict[str, Any]:
+        url = f"{cls.LLAMA_CLOUD_URL}/files/{file_id}/content?project_id={project_id}"
+        return cls.make_request(url)
+
+    @classmethod
+    def download_file(cls, url: str, local_file_path: str):
+        logger.info(f"Downloading file to {local_file_path}")
+        # Create directory if it doesn't exist
+        os.makedirs(cls.LOCAL_STORE_PATH, exist_ok=True)
+        # Download the file
+        with requests.get(url, stream=True) as r:
+            r.raise_for_status()
+            with open(local_file_path, "wb") as f:
+                for chunk in r.iter_content(chunk_size=8192):
+                    f.write(chunk)
+        logger.info("File downloaded successfully")
+
+    @classmethod
+    def task_download_llamacloud_pipeline_file(
```

Comment:
doesn't seem to have a dependency to `BackgroundTasks` so we could also call this as a normal method right? so basically, we can see this as a service method